### ISSUE TITLE
VSCodium installation creates Microsoft/DeveloperTools/deviceid file in ~/Library/Application\ Support (macOS)

### ISSUE THREAD

**Issue by cspence001** [VSCodium/vscodium#1960]
**Describe the bug**
Upon installation of VSCodium  - a Microsoft directory is created in ~/Library/Application\ Support, with a unique deviceid file. 

**Please confirm that the issue/resolution isn't already documented**
- [X ] I checked [the Docs page](https://github.com/VSCodium/vscodium/blob/master/docs/index.md) and [the Troubleshooting page](https://github.com/VSCodium/vscodium/blob/master/docs/troubleshooting.md) and my issue is not mentioned there.

**To Reproduce**
1. Install VSCodium
2. From Terminal:
`open -a Finder ~/Library/Application\ Support/Microsoft/DeveloperTools/deviceid`
or
`cat ~/Library/Application\ Support/Microsoft/DeveloperTools/deviceid`

**Expected behavior**
Considering VSCodium's intent to remove Microsoft's telemetry, the creation of a Microsoft directory with a unique device ID in ~/Library/Application Support/Microsoft/DeveloperTools/deviceid on macOS for VSCodium is an unexpected behavior.

**Desktop (please complete the following information):**
 - OS: [MacOS Ventura 13.6.7]
 - Architecture [Darwin x64 22.6.0]
 - Version [1.91.1]
 - App Manager [None]
 - Sandboxed [Yes]

**Additional context**
- I don't remember having encountered the creation of this directory/file in previous installations of VSCodium
- "crash-reporter-id" in ~/.vscode-oss/argv.json displays different unique id than the one shown in  ~/Library/Application Support/Microsoft/DeveloperTools/deviceid
- Deletion of this file appears to not have an effect on application function, or installation of extensions, etc. 
- It does not respawn upon re-initialization of VSCode. However, it has not been tested as to whether or not updating the application will respawn the file.

**Comment by daiyam** [comment]
Mine was created on the 4th June but maybe by VSCode.
Updating VSCodium or restarting it don't regenerate the file.
Updating VSCode or restarting it don't regenerate the file.

The package [`@vscode/deviceid`](https://www.npmjs.com/package/@vscode/deviceid) and the function [`getdevDeviceId()`](https://github.com/search?q=repo%3Amicrosoft%2Fvscode+getdevDeviceId&type=code) are responsible for creating the file.

**Comment by cspence001** [comment]
> The package [`@vscode/deviceid`](https://www.npmjs.com/package/@vscode/deviceid) and the function [`getdevDeviceId()`](https://github.com/search?q=repo%3Amicrosoft%2Fvscode+getdevDeviceId&type=code) are responsible for creating the file.

Yes I have seen this, as well as citation to its use-case as a [developer telemetry API ](https://github.com/microsoft/vscode-deviceid/blob/main/.config/TSAOptions.json). This is where the ms node creates the dir [@vscode-deviceid/src/storage.ts](https://github.com/microsoft/vscode-deviceid/blob/main/src/storage.ts). The package looks to be relatively new within the last 2-3 months.
    
Notes:
- ~/Library/Application\ Support/VSCodium/User/globalStorage/storage.json contains:
    "telemetry.machineId": "$IDval",
    "telemetry.devDeviceId": "$IDval"
- telemetry.devDeviceId $IDval matches the value found in ~/Library/Application Support/Microsoft/DeveloperTools/deviceid
- telemetry.machineid $IDval does not match ~/Library/Application\ Support/VSCodium/machineid
